src/effects/effectslot.cpp

```diff
@@ -178,16 +179,20 @@ EffectManifestPointer EffectSlot::getManifest() const {
     return m_pManifest;
 }
 
-void EffectSlot::reload(const QSet<ChannelHandleAndGroup>& activeInputChannels) {
-    loadEffect(m_pManifest,
-            m_pEffectsManager->createProcessor(m_pManifest),
-            activeInputChannels);
+void EffectSlot::reload() {
+    for (auto const& pParameterSlot : m_knobParameterSlots) {
+        pParameterSlot->reload();
+    }
+    for (auto const& pParameterSlot : m_buttonParameterSlots) {
+        pParameterSlot->reload();
+    }
```

Comment:
There are several places in EffectSlot where code is duplicated for knob and button parameters. It would help to store a QList<EffectParameterSlotBasePointer> and iterate over that instead.